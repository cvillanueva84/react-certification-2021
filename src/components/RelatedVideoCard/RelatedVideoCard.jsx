import React from 'react';
import { Link } from 'react-router-dom';
import { Paper } from '@material-ui/core';

import {
  RelatedVideo,
  RelatedVideThumbnail,
  RelatedVideoTitle,
} from './RelatedVideoCard.modules';
import './RelatedVideoCard.styles.css';

function RelatedVideoCard(props) {
  const { videoId, title, imageUrl } = props;

  return (
    <Link to={`/video/${videoId}`} key={videoId}>
      <Paper className="related-video-card" elevation={3}>
        <RelatedVideo>
          <RelatedVideThumbnail>
            <img alt={title} src={imageUrl} />
          </RelatedVideThumbnail>
          <RelatedVideoTitle>{title}</RelatedVideoTitle>
        </RelatedVideo>
      </Paper>
    </Link>
  );
}

export default RelatedVideoCard;
