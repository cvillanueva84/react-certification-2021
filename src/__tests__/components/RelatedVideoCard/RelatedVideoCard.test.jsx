import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import Adapter from 'enzyme-adapter-react-16';
import EnzymeToJson from 'enzyme-to-json';
import { mount, configure } from 'enzyme';

import RelatedVideoCard from '../../../components/RelatedVideoCard';

describe('RelatedVideoCard', () => {
  let relatedVideoCard;
  beforeEach(() => {
    configure({ adapter: new Adapter() });
    relatedVideoCard = mount(
      <BrowserRouter>
        <RelatedVideoCard />
      </BrowserRouter>
    );
  });

  it('Renders VideoDetails Page', () => {
    expect(EnzymeToJson(relatedVideoCard)).toMatchSnapshot();
  });
});
