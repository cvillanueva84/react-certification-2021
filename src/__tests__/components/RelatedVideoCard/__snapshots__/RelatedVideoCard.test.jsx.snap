// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedVideoCard Renders VideoDetails Page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RelatedVideoCard>
      <Link
        to="/video/undefined"
      >
        <LinkAnchor
          href="/video/undefined"
          navigate={[Function]}
        >
          <a
            href="/video/undefined"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(Paper))
              className="related-video-card"
              elevation={3}
            >
              <ForwardRef(Paper)
                className="related-video-card"
                classes={
                  Object {
                    "elevation0": "MuiPaper-elevation0",
                    "elevation1": "MuiPaper-elevation1",
                    "elevation10": "MuiPaper-elevation10",
                    "elevation11": "MuiPaper-elevation11",
                    "elevation12": "MuiPaper-elevation12",
                    "elevation13": "MuiPaper-elevation13",
                    "elevation14": "MuiPaper-elevation14",
                    "elevation15": "MuiPaper-elevation15",
                    "elevation16": "MuiPaper-elevation16",
                    "elevation17": "MuiPaper-elevation17",
                    "elevation18": "MuiPaper-elevation18",
                    "elevation19": "MuiPaper-elevation19",
                    "elevation2": "MuiPaper-elevation2",
                    "elevation20": "MuiPaper-elevation20",
                    "elevation21": "MuiPaper-elevation21",
                    "elevation22": "MuiPaper-elevation22",
                    "elevation23": "MuiPaper-elevation23",
                    "elevation24": "MuiPaper-elevation24",
                    "elevation3": "MuiPaper-elevation3",
                    "elevation4": "MuiPaper-elevation4",
                    "elevation5": "MuiPaper-elevation5",
                    "elevation6": "MuiPaper-elevation6",
                    "elevation7": "MuiPaper-elevation7",
                    "elevation8": "MuiPaper-elevation8",
                    "elevation9": "MuiPaper-elevation9",
                    "outlined": "MuiPaper-outlined",
                    "root": "MuiPaper-root",
                    "rounded": "MuiPaper-rounded",
                  }
                }
                elevation={3}
              >
                <div
                  className="MuiPaper-root related-video-card MuiPaper-elevation3 MuiPaper-rounded"
                >
                  <styled.div>
                    <div
                      className="sc-bdnxRM gdUVKc"
                    >
                      <styled.div>
                        <div
                          className="sc-gtsrHT fTcaF"
                        >
                          <img />
                        </div>
                      </styled.div>
                      <styled.div>
                        <div
                          className="sc-dlnjwi edrUSp"
                        />
                      </styled.div>
                    </div>
                  </styled.div>
                </div>
              </ForwardRef(Paper)>
            </WithStyles(ForwardRef(Paper))>
          </a>
        </LinkAnchor>
      </Link>
    </RelatedVideoCard>
  </Router>
</BrowserRouter>
`;
